src/H5FDs3comms.c

```diff
@@ -1910,13 +1984,20 @@ H5FD_s3comms_load_aws_profile(const char *profile_name, char *key_id_out, char *
         if (H5FD__s3comms_load_aws_creds_from_file(
                 credfile, profile_name, (*key_id_out == 0) ? key_id_out : NULL,
                 (*secret_access_key_out == 0) ? secret_access_key_out : NULL,
+                (*session_token_out == 0) ? session_token_out : NULL,
                 (*aws_region_out == 0) ? aws_region_out : NULL) == FAIL)
             HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "unable to load from aws config");
         if (fclose(credfile) == EOF)
             HGOTO_ERROR(H5E_FILE, H5E_CANTCLOSEFILE, FAIL, "unable to close config file");
         credfile = NULL;
     } /* end if credential file opened */
 
+    /* Check for credentials in environment variables.  Environment variables will override
+     * credentials from credentials/config files and just load them if there were none in
+     * the files. */
+    ret_value = H5FD__s3comms_load_aws_creds_from_env(key_id_out, secret_access_key_out, session_token_out,
```

Comment:
It's fine not use `HGOTO_ERROR` here right now since it's at the end of the routine anyway, but it would be good to explicitly check `ret_value` in case that changes later.